source/common/json/config_schemas.cc

```diff
@@ -677,3 +675,285 @@ const std::string Json::Schema::ROUTER_HTTP_FILTER_SCHEMA(R"EOF(
     "additionalProperties" : false
   }
   )EOF");
+
+const std::string Json::Schema::CLUSTER_MANAGER_SCHEMA(R"EOF(
+  {
+    "$schema": "http://json-schema.org/schema#",
+    "definitions" : {
+      "sds" : {
+        "type" : "object",
+        "properties" : {
+          "cluster" : {"type" : "object"},
+          "refresh_delay_ms" : {
+            "type" : "integer",
+            "minimum" : 0,
+            "exclusiveMinimum" : true
+          }
+        },
+        "required" : ["cluster", "refresh_delay_ms"],
+        "additionalProperties" : false
+      },
+      "cds" : {
+        "type" : "object",
+        "properties" : {
+          "cluster" : {"type" : "object"},
+          "refresh_interval_ms" : {
+            "type" : "integer",
+            "minimum" : 0,
+            "exclusiveMinimum" : true
+          }
+        },
+        "required" : ["cluster"],
+        "additionalProperties" : false
+      }
+    },
+    "properties" : {
+      "clusters" : {
+        "type" : "array",
+        "items" : {"type": "object"}
```

Comment:
every element is a cluster, can you enforce checking for cluster definition here?